hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/blockmanagement/BlockManagerSafeMode.java

```diff
@@ -640,7 +643,17 @@ private void reportStatus(String msg, boolean rightNow) {
    */
   private class SafeModeMonitor implements Runnable {
     /** Interval in msec for checking safe mode. */
-    private static final long RECHECK_INTERVAL = 1000;
+    private long recheckInterval = 1000;
+
+    public SafeModeMonitor() {
+      Configuration conf = new HdfsConfiguration();
+      long interval = conf.getLong(
+              DFS_NAMENODE_SAFEMODE_RECHECK_INTERVAL_KEY,
+              DFS_NAMENODE_SAFEMODE_RECHECK_INTERVAL_DEFAULT);
+      if (interval > 0) {
+        recheckInterval = interval;
```

Comment:
Add a warn log telling the configured value is invalid and you are using the default value.

Secondly as of now you are having two variables, `recheckInterval` & `interval`, Keep only `recheckInterval` and initialise it in the constructor, if the value is greater than 0, no issues, if it is not set it to `DFS_NAMENODE_SAFEMODE_RECHECK_INTERVAL_DEFAULT`

Reply:
@ayushtkn , thank you very much for your suggestions, I will submit some improved codes later.